pkg/ipamfetcher/customer.go

```diff
@@ -64,11 +72,50 @@ func (d *Device42CustomerFetcher) FetchCustomers(ctx context.Context) ([]domain.
 	}
 	customers := make([]domain.Customer, 0, len(getCustomersResponse.Customers))
 	for _, customer := range getCustomersResponse.Customers {
+		businessUnit := customer.CustomFields.GetValue("Description")
+		if businessUnit == "" {
+			// fallback to customer name
+			businessUnit = customer.Name
+		}
 		customers = append(customers, domain.Customer{
 			ID:            strconv.Itoa(customer.ID),
-			ResourceOwner: customer.ContactInfo,
-			BusinessUnit:  customer.CustomFields.GetValue("Description"),
+			ResourceOwner: getResourceOwner(customer),
+			BusinessUnit:  businessUnit,
 		})
 	}
 	return customers, nil
 }
+
+func getResourceOwner(customer customer) string {
+
+	/*
+	   Find the best resource owner by searching through the contacts
+	   for non-empty Value for the following Key, in this priority order:
+	   1. Team Lead Contact Email
+	   2. Administrative Contact Email
+	   3. SRE Contact Email
+	   4. Technical Contact Email
+	   5. Fall back to The "contact_info" field directly on the customer record
+	*/
+
+	owner := customer.ContactInfo
+	fromPriority := 5
+	for _, contact := range customer.Contacts {
```

Comment:
Question: Is there a better way to right this instead of a bunch of if statements? I can't think of a way, but it feels very redundant.

Reply:
I thought it was pretty clever, actually.

Reply:
You could do a case/switch with fall-through. Might be slightly more readable.